### PR TITLE
Move `ExecutionParameters` to a seperate file aftermath

### PR DESCRIPTION
**Pull request by Jacfomg**
See https://github.com/qiboteam/qibolab/pull/502.

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,9 +16,9 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.12"
-qibolab = { git = "https://github.com/qiboteam/qibolab.git", branch = "main" }
+qibolab = { git = "https://github.com/qiboteam/qibolab.git", branch = "move-execution-parameters" }
 qibo = "^0.1.9"
-numpy = "^1.25.0"
```
Comment (andrea-pasquale) [inline pyproject.toml:21]: Is there a specific reason for this downgrade?

**Review by andrea-pasquale — APPROVED** [review]
Thanks @Jacfomg, LGTM.
Remember to upgrade the lock after we merge the corresponding PR in qibolab.

File: pyproject.toml (hunk 1 above)
Comment (Jacfomg) [reply pyproject.toml:21]: Yup, https://github.com/qiboteam/qibocal/pull/398#issuecomment-1601276341

Comment (andrea-pasquale) [reply pyproject.toml:21]: I see. Just to make sure that we don't forget could you please add a comment above this line linking @AleCandido 's comment?

**Review by AleCandido — APPROVED** [review]
Thanks for this.

Just wait for Qibolab PR to be merged, and please remember to update the Qibolab dependency before merging.

**Pull request closed without merging**